### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by rugk** [nvaccess/nvda#6104]
I am a bit surprised to not see any issue about this, but you only support Windows curreently.
As you use Qt you might also support other OSes, such as Linux, in an easy way - at least the UI would be easy.

**Comment by josephsl** [comment]
Hi,

A few clarifications:

* NVDA uses wxWidgets toolkit (technically, wxPython). Due to accessibility issues in earlier versions of QT, NV Access has decided to use wxWidgets for now.
* _Porting to Linux means using Linux kernel API and assistive API's provided by Linux, and this is a significant effort.
* This not only affects NVDA screen reader itself, but add-ons - some add-ons assume that we're using Win32 API.

Closing as this is not feasible at the moment. Thanks.

**Comment by rugk** [comment]
Okay, so it is a least good to know the reason for Windows-only compatibility.

**Comment by derekriemer** [comment]
It should be noted that there's a good screen reader on linux called 
orca. If you need a screen reader for linux, I'd recommend you use it, 
it comes with gnome, to start it, I believe you'd press super+alt+s.


On 6/22/2016 3:44 PM, rugk wrote:
>
> Okay, so it is a least good to know the reason for Windows-only 
> compatibility.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/6104#issuecomment-227886477>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivcKgCfdjbyZQzC8qZhXp-nY51udfks5qOazTgaJpZM4I8OQ5>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

**Comment by zahra21** [comment]
hi.
i also decided to migrate to linux, since i am windows xp user and dont know anything about terminal and coding, i decided to use wine to use my windows programs on linux.
i have less than ten important programs and realized that wine supports them.
the only issue is nvda.
i realy miss nvda, love it and dont want to lose it!
i used nvda for many years, know many people and many know me, asked and answered thousands of question in the users mailing list.
can someone provide me a solution how to use nvda on linux?
i also told my request in wine bug report website too.

**Comment by rugk** [comment]
Okay, so did you try out Orca as recommended in this issue previously?☺

**Comment by zahra21** [comment]
no, i did not, as i said, i realy love nvda and dont want to miss it.
moreover, for me a novice user without any informations about coding and using terminal, the only solution is wine.
unfortunately nvda is not supported well with wine and even orca does not work with wine.
it means that using windows softwares with wine is not possible for blind users!
see these bugs in wine bug report website.

http://bugs.winehq.org/show_bug.cgi?id=18905

https://bugs.winehq.org/show_bug.cgi?id=37681

wine supports more than 25000 softwares, if i realized correctly, but blind users are deprived from this great feature!
i commented in one of the above link and i should add in second report too.

**Comment by rugk** [comment]
Orca does not need wine, it is integrated into Linux distros.

A real support for NVDA on Linux would certainly better than wine stuff. So maybe better urge the NVDA devs to re-open this issue and at least consider implementing Linux support.

**Comment by zahra21** [comment]
but i want to exactly use my exe versions of my current softwares on linux, but wine is not accessible for blind users!
i wish that wine support accessibility features of windows that i can install first nvda and then other my exe files just like my current windows xp version.
add supporting of nvda for linux is good also, but i dont want terminal and wish that install all of my softwares like windows just with next next!

**Comment by Brian1Gaff** [comment]
Well you cannot its a totally different system, that can look like Windows, 
that is all.

There have been attempts in the past to make Windows clones, but since the 
code Windows uses is copyrighted, its nearly always a bad compromise at 
best.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by zstanecic** [comment]
Hi Zahra,
That what you want is so irreal.
Wine was never fully accessible and will ot be.
They have problems with the acessibility from the very beginning
Wysłane z aplikacji Poczta dla Windows 10

Od: zahra21
Wysłano: środa, 25 kwietnia 2018 10:02
Do: nvaccess/nvda
DW: Subscribed
Temat: Re: [nvaccess/nvda] Linux support (#6104)

but i want to exactly use my exe versions of my current softwares on linux, but wine is not accessible for blind users!
i wish that wine support accessibility features of windows that i can install first nvda and then other my exe files just like my current windows xp version.
add supporting of nvda for linux is good also, but i dont want terminal and wish that install all of my softwares like windows just with next next!

**Comment by zahra21** [comment]
hi zstanecic.
but if developers attention to accessibility and specially provide accessibility features of windows on linux wine, blind users can easily use nvda and they can install windows programs on linux!

i read this bug against wine and it is currently active and users can comment.
from the last comment, i realized that previous versions of nvda can be install on wine if i am correct!

http://bugs.winehq.org/show_bug.cgi?id=18905

**Comment by zstanecic** [comment]
Yes, but it is unreliable.
It has a problem with the focus

Wysłane z aplikacji Poczta dla Windows 10

Od: zahra21
Wysłano: środa, 25 kwietnia 2018 10:53
Do: nvaccess/nvda
DW: zstanecic; Comment
Temat: Re: [nvaccess/nvda] Linux support (#6104)

hi zstanecic.
but if developers attention to accessibility and specially provide accessibility features of windows on linux wine, blind users can easily use nvda and they can install windows programs on linux!
i read this bug against wine and it is currently active and users can comment.
from the last comment, i realized that previous versions of nvda can be install on wine if i am correct!
http://bugs.winehq.org/show_bug.cgi?id=18905

**Comment by rugk** [comment]
Then maybe try out a previous version.
My point still was that native support will always be better than wine and such things. And theoretically it is possible as the first reply to this issue shows.
Of course, I understand that running nvda via wine would be the easier short-term solution and maybe currently preferable,  but the long-term solution would be that nvda can run exactly the same as it can run on windows. So that you have the same features and so on.

**Comment by zahra21** [comment]
i also vote for adding nvda into linux. i realy love nvda and dont want to miss it!

**Comment by josephsl** [comment]
Hi,

Although user experience is a valid claim for letting NVDA be ported to Linux, it cannot be done easily. As I and others said above, it requires significant redesign so NVDA can use user-mode Linux API's and accessibility framework in place. Complicating matters is the fact that there are thousands of Linux distributions, so if we do target a specific distro, it may raise concerns about bias and other factors later. In other words, this ticket shall remain closed until the day it becomes possible to do something about it, especially given vast internal and attitude differences between operating systems.

Thanks.

**Comment by zahra21** [comment]
hi joseph.
i read hundreds of articles about linux distros and realized that orca is installed in many linuxes that i studied about them!
they are many graphical desktop environments on linux which are used in different versions of linux, but all of them which i studied there informations, use orca screen reader!
which means that linux has one accessibility api that you can implement and support nvda in all of accessible linux versions!
moreover, you can support thousands of distros of linux, beside some other operating systems!
solaris, bsd, unix and linux all use the same accessibility api!
in general, i read hundreds of articles about linux both in farsi and english languages and unfortunately, i forgot the name of accessibility api common between on linux, bsd, unix and solaris now!

**Comment by derekriemer** [comment]
I'm locking this ticket to prevent lots of chatter. This simply is impossible, because screen readers deal with the operating system at low levels and providing a portable cross OS operating system is both infeasible and not practical. There are lots of things going on behind the scenes.
